feat: 189-storage-device-details-of-enlisted-machines

tobiasdemendonca opened the pull request:
## Description of changes

Add storagee device details to `maas_machine`. 

## Issue or ticket link (if applicable)

Resolves: #189

## Checklist

- [x] I have written a PR title that follows the advice in DEVELOPMENT.md with the title format `type(scope): title`
- [x] I have written a description of the changes in the PR that is clear and concise.
- [x] I have updated the documentation to reflect the changes.
- [ ] I have added or updated the tests to reflect the changes.
- [x] I have run the unit tests and they pass.
- [ ] I have run the acceptance tests and they pass.

skatsaounis commented on maas/resource_maas_machine.go:
```diff
@@ -68,6 +69,35 @@ func resourceMAASMachine() *schema.Resource {
+						"id_path": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID path of the block device. This should be a path that is fixed and doesn't change depending on the boot order or kernel version.",
```
The description here can confuse the user since this field is computed. "This should be a path ....". Most probably it was copied from `maas_block_device` resource.

skatsaounis commented on maas/resource_maas_machine.go:
```diff
@@ -68,6 +69,35 @@ func resourceMAASMachine() *schema.Resource {
+						"model": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Model of the block device. Used in conjunction with `serial` argument. Conflicts with `id_path`. This argument is computed if it's not given.",
```
Same thing here. Let's try to change both descriptions so that they describe the field but as something computed, not specified by the user.

skatsaounis reviewed (changes requested):
I added some comments for improving the description of the fields. In addition, I do not know if it is a bug or not, but I spotted that the Markdown Does not contain descriptions of the read-only fields.

Finally, given that we do not have yet a `maas_machine` acceptance test we have to decide whether it is a good time to introduce one with this PR or avoid it.

tobiasdemendonca commented:
> I added some comments for improving the description of the fields. In addition, I do not know if it is a bug or not, but I spotted that the Markdown Does not contain descriptions of the read-only fields.
> 
> Finally, given that we do not have yet a `maas_machine` acceptance test we have to decide whether it is a good time to introduce one with this PR or avoid it.

I thought we discussed not including the maas machine test for this PR but I could be wrong. I do think we probably need one, but I guess the setup would be a bit complex if I were to mimic what we discussed a few days ago which involved creating a vm, renaming it outside of maas, removing the old resource from maas, and then using the `maas_machine` to create and commission it again. 

I might suggest adding that in a separate PR because of this, what do you think? Ultimately though I can prioritise this sprint or next sprint if we think it's really required: just because it's a bit harder to implement doesn't mean we shouldn't add it! But I think it could be considered related but separate to this PR.

tobiasdemendonca commented:
Okay I found an existing issue, doesn't look like this will be fixed soon https://github.com/hashicorp/terraform-plugin-docs/issues/28 I can try and work out how to add this in manually

skatsaounis reviewed (approved):
No reason to add things manually for now. The PR LGTM +1. Try to rebase, then make the CI happy and finally merge it :beers:

tobiasdemendonca commented:
I believe the best way is with a template, I implemented it but as soon as you want to adjust the schema, you have to write it manually so there's a risk of it getting out of sync. I've reverted my changes for now but can discuss this when we're back if needed

SK1Y101 reviewed (dismissed):
+1 from me too

skatsaounis reviewed (approved):
LGTM +1
